tests/test_config_entries.py

```diff
@@ -3313,7 +3314,19 @@ async def test_reauth(hass):
 
     assert mock_init.call_args.kwargs["data"]["extra_data"] == 1234
 
+    assert entry.entry_id != entry2.entry_id
+
     # Check we can't start duplicate flows
     entry.async_start_reauth(hass, {"extra_context": "some_extra_context"})
     await hass.async_block_till_done()
-    assert len(flows) == 1
+    assert len(hass.config_entries.flow.async_progress()) == 1
+
+    # Check we can't start duplicate when the context context is different
```

Comment:
when the context is different

Reply:
https://github.com/home-assistant/core/pull/81789